draft-ietf-tls-esni.md

```diff
@@ -388,53 +388,53 @@ may use the "outer_extensions" extension.
        outer_extensions(0xff04), (65535)
     } ExtensionType;
 
-    struct {
-       ExtensionType outer_extensions<2..254>;
-       uint8 inner_digest<32..255>;
-    } OuterExtensions;
+    ExtensionType OuterExtensions<2..254>;
 ~~~
 
 OuterExtensions consists of one or more ExtensionType values, each of which
-reference an extension in ClientHelloOuter, and a digest of the complete
-ClientHelloInner.
+reference an extension in ClientHelloOuter.
 
 When sending ClientHello, the client first computes ClientHelloInner, including
 any PSK binders. It then computes a new value, the EncodedClientHelloInner, by
-first making a copy of ClientHelloInner. It then MAY substitute extensions
-which it knows will be duplicated in ClientHelloOuter. To do so, the client
-computes the digest of the entire ClientHelloInner message as:
-
-~~~
-  inner_digest = Expand(Extract("", inner),
-                        "tls13 ech inner digest", Nh)
-~~~
-
-where `inner` is the ClientHelloInner structure and `Extract`, `Expand`, and
-`Nh` are as defined by the KDF API in {{!I-D.irtf-cfrg-hpke}}. Then, the client
-removes and replaces extensions from EncodedClientHelloInner with a single
-"outer_extensions" extension. Removed extensions MUST be ordered consecutively
-in ClientHelloInner. The list of outer extensions,
-OuterExtensions.outer_extensions, includes those which were removed from
+first making a copy of ClientHelloInner. It then replaces the legacy_session_id
+field with an empty value.
```

Comment:
The empty string. I previously wrote "clears the value", but @cjpatton objected to that one. :-P How about s/empty value/empty string/? We haven't introduced a new encoded structure yet (https://github.com/tlswg/draft-ietf-tls-esni/pull/336#issuecomment-707341408), so we need to put *something* in there.

Reply:
So a zero-length string, as opposed to a 32 byte string of zeroes? (Whatever makes that more clear is fine by me.)

Reply:
I mean the point is to encode fewer bytes, so yeah. I feel like, given the field is defined by `opaque legacy_session_id<0..32>`, it would be unreasonable to interpret "empty value" or "empty string" as a 32-byte string of zeros. :-P

I dunno, I think all the options so far have been pretty clear, so I'm not the best judge here. The intent is to stick the zero-length string in there. Which one do you prefer?
* clears the legacy_session_id field
* replaces the legacy_session_id field with an empty value
* replaces the legacy_session_id field with a zero-length value
* replaces the legacy_session_id field with an empty string
* replaces the legacy_session_id field with a zero-length string

Reply:
+1 "replaces the legacy_session_id field with an empty string"

Reply:
Okay, empty string it is. I don't care, just pick one. :-)

Reply:
Done.